.github/workflows/check_rebuild_tempest.yaml

```diff
@@ -0,0 +1,66 @@
+name: Check and Rebuild Tempest Image
+
+on:
+  workflow_call:
+    inputs:
+      branch:
+        description: The branch to run checks on and re-build images from
+        type: string
+        required: true
+
+jobs:
+  check-update:
+    runs-on: ubuntu-latest
+    outputs:
+      rebuild_image: ${{ steps.check.outputs.rebuild_image }}
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          ref: ${{ inputs.branch }}
+      - name: Install yq
+        run: |
+          sudo snap install yq
+      - name: Check if a recent snap updates is found
+        id: check
+        run: |
+          # get the snap name and its channel that the ROCK image should track
+          snap=$(yq '.parts.[].stage-snaps.[]' rocks/tempest/rockcraft.yaml)
+          IFS='/' read -r name channel <<< "$snap"
+          echo "snap: $name, channel: $channel"
+
+          # parse date of the last release and reformat it to seconds
+          format='+%s'
+          last_release_date=$(snap info tempest | grep "2023.2/stable" | awk '{ print $3 }')
+          echo "last release date in channel $channel: $last_release_date"
+          last_release_date_in_seconds=$(date -d "$last_release_date" "$format")
+
+          # confirm the validity of the release date
+          if [[ $? != 0 ]]; then
+            echo "$last_release_date is not a valid date."
+          fi
+
+          # get current date and reformat it to seconds
+          cur_date=$(date)
+          echo "current date: $cur_date"
+          cur_date_in_seconds=$(date -d "$cur_date" "$format")
+
+          # get the date difference between two days
+          date_diff=$(( ($cur_date_in_seconds - $last_release_date_in_seconds)/(60*60*24) ))
+          
+          # re-build image if the date difference is less than or equal to 5 days
```

Comment:
If the period of `update_tempest_releases.yaml` is larger than 5 days, it won't have the issue. Currently. it seems that it will have the issue you mentioned.

Reply:
e.g. `0 0 */5.5 * *`

Reply:
`0 0 */5.5 * *` would also have multi-run problem at the end/beginning of a month (because the third field of cron schedule is `day in month`). An example is explained [here](https://stackoverflow.com/a/55087873). I chose specific days in week to have a more predictable behavior. 

Though I don't think re-running workflow is a big issue, I agree comparing image update date with snap update date would be a better solution. The reason I didn't go for it was because I wasn't able to find the upstream packages through a standard GitHub API (e.g. ). I will research about this further.

Reply:
Interesting, I thought every * days should not be dependent on month, thanks for the info!